tests/test_git_snapshot.py

```
@@ -260,6 +260,56 @@ def test_unstaged_intent_to_add_and_rename_have_exact_identities(tmp_path):
 
 
+def test_unstaged_includes_non_ignored_untracked_files_with_exact_identities(tmp_path):
+    repo = make_repo(tmp_path)
+    write(repo, ".gitignore", b"ignored.py\n")
+    write(repo, "tracked.py", b"def tracked():\n    return 1\n")
+    commit_all(repo)
+
+    write(repo, "new.py", b"def added():\n    return 1\n")
+    write(repo, "ignored.py", b"def ignored():\n    return 1\n")
+    write(repo, "bin/tool", b"#!/bin/sh\nexit 0\n")
+    os.chmod(repo / "bin/tool", 0o755)
```

Ruff (0.16.1)

[error] `os.chmod` setting a permissive mask `0o755` on file or directory

(S103)